File: DotNet/SETPractice/DesignPattern/FacadePattern/Menu.cs
namespace DesignPattern.FacadePattern
{
    public class Menu : IMenu
    {
        public List<string> GetNonVegMenu()
        {
            return Enum.GetValues(typeof(VegMenu))
                .Cast<VegMenu>()
                .Select(v => v.ToString())
                .ToList();
        }

        public List<string> GetVegMenu()
        {
            return Enum.GetValues(typeof(NonVegMenu))
                .Cast<NonVegMenu>()
                .Select(v => v.ToString())
                .ToList();
        }

        public List<string> GetAllMenu()
        {
            List<string> menu = GetVegMenu();
            List<string> nonVegMenu = GetNonVegMenu();
            menu.AddRange(nonVegMenu);
            return menu.ToList();
        }
    }
}

File: Java/SETPractice/src/main/java/DesignPattern/FacadePattern/Menu.java
package DesignPattern.FacadePattern;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class Menu implements IMenu{
    @Override
    public List<String> GetVegMenu() {
        return Stream.of(VegMenu.values())
                .map(Enum::name)
                .collect(Collectors.toList());
    }

    @Override
    public List<String> GetNonVegMenu() {
        return Stream.of(NonVegMenu.values())
                .map(Enum::name)
                .collect(Collectors.toList());
    }

    @Override
    public List<String> GetAllMenu() {
        List<String> menu = GetVegMenu();
        menu.addAll(GetNonVegMenu());
        return menu;
    }
}
